src/tinuous/util.py

```diff
@@ -46,17 +47,78 @@ def iterfiles(dirpath: Path) -> Iterator[Path]:
                 yield p
 
 
+class LazySlicingFormatter(Formatter):
```

Comment:
Please add a docstring describing its "Function in life" (ref: https://www.youtube.com/watch?v=uLx7-jrji9o)